transaction/src/main/java/org/wildfly/httpclient/transaction/ClientHandlers.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2024 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.httpclient.transaction;
+
+import static org.wildfly.httpclient.transaction.ByteOutputs.byteOutputOf;
+import static org.wildfly.httpclient.transaction.Serializer.deserializeXid;
+import static org.wildfly.httpclient.transaction.Serializer.deserializeXidArray;
+import static org.wildfly.httpclient.transaction.Serializer.serializeXid;
+
+import io.undertow.client.ClientResponse;
+import org.jboss.marshalling.ByteInput;
+import org.jboss.marshalling.ByteOutput;
+import org.jboss.marshalling.InputStreamByteInput;
+import org.jboss.marshalling.Marshaller;
+import org.jboss.marshalling.Unmarshaller;
+import org.wildfly.httpclient.common.HttpTargetContext;
+import org.wildfly.httpclient.common.NoFlushByteOutput;
+import org.xnio.IoUtils;
+
+import javax.transaction.xa.Xid;
+import java.io.Closeable;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
```

Comment:
Comments would help here to describe the overall purpose of this class in the request/response interaction, something like:
"This class defines helper methods for marshaling common transaction invocation request parameter types, as well as handling the common transaction invocation response result types"

Reply:
How about: "Provides factory methods for client-side handlers of the TXN over HTTP protocol." ?